### PR TITLE
Add flow for self diagnosis report type

### PR DESCRIPTION
**Pull request by baugztar**
* Add support for new report-type `SelfDiagnosis`
* Users authenticating as underaged now get shown a dedicated error screen
* Add temporary texts for underaged screen

### CODE REVIEW

File: NDB.Covid19/NDB.Covid19.iOS/Views/ErrorStatus/ErrorPageViewController.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/ErrorStatus/ErrorPageViewController.cs:114]: What about the situations when under 16yo logs in with lookup in MSIS? Shouldn't the app then show the `REGISTER_ERROR_NOMATCH_HEADER` error page but without `ContinueWithSelftestButton` button?

File: NDB.Covid19/NDB.Covid19/Enums/ReportInfectedType.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/Enums/ReportInfectedType.cs]: Isn't  this enum already defined? Check out `NDB.Covid19.ProtoModels.TemporaryExposureKey.Types`

File: NDB.Covid19/NDB.Covid19/Models/PersonalDataModel.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/Models/PersonalDataModel.cs]: Re-write with string interpolation: `$"{nameof(PersonalDataModel)}......"`

File: NDB.Covid19/NDB.Covid19/Models/PersonalDataModel.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/Models/PersonalDataModel.cs]: Re-write with string interpolation

File: NDB.Covid19/NDB.Covid19/Models/SelfDiagnosisSubmissionDTO.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/Models/SelfDiagnosisSubmissionDTO.cs:34]: Did you check a scenario that if the person has at some point in the past reported infected with lookup in MSIS and decided to share the keys with EU, then this will be already set to true, so if the same user reports infected with self-test, the users' keys will be marked for sharing with EU?

**Review by pavshr — CHANGES_REQUESTED** [review]
Very nice, some small things need to be addressed before approval.

**Comment by pavshr** [comment]
@baugztar, pls also extend Developer tools to include the following output for sent keys:
1. Add `ReportType` that is set in DTO
2. Add the value of `IsSharingAllowed` of the DTO, if it is not there yet

File: NDB.Covid19/NDB.Covid19.iOS/Views/ErrorStatus/ErrorPageViewController.cs
Comment (baugztar) [reply NDB.Covid19/NDB.Covid19.iOS/Views/ErrorStatus/ErrorPageViewController.cs:114]: The `no-msis` claim is not included in the token from either flows, as the underage check is done on the Verification server before any MSIS lookup is done. So in this change the underage error screen will be the same for both scenarios (no continue button).

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/InformationAndConsentActivity.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/InformationAndConsentActivity.cs:37]: Looks a bit weird, maybe import whole `using NDB.Covid19.ProtoModels.TemporaryExposureKey.Types` with static, e.g., like on the line 17?

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/InformationAndConsentViewController.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/InformationAndConsentViewController.cs]: same as above, import as static to just refer to `ReportType` in the code so it looks cleaner

File: NDB.Covid19/NDB.Covid19/Models/SelfDiagnosisSubmissionDTO.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/Models/SelfDiagnosisSubmissionDTO.cs]: Please do it in a cleaner way: `(int)ReportType.SelfReport`

File: NDB.Covid19/NDB.Covid19/ViewModels/ENDeveloperToolsViewModel.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/ViewModels/ENDeveloperToolsViewModel.cs:81]: Great, much appreciated by test team :)

File: NDB.Covid19/NDB.Covid19/ViewModels/InformationAndConsentViewModel.cs
Comment (pavshr) [inline NDB.Covid19/NDB.Covid19/ViewModels/InformationAndConsentViewModel.cs:105]: Is it also reset in the preferences as well?

**Review by pavshr — COMMENTED** [review]
Good, some last thing :)

File: NDB.Covid19/NDB.Covid19/ViewModels/InformationAndConsentViewModel.cs
Comment (baugztar) [reply NDB.Covid19/NDB.Covid19/ViewModels/InformationAndConsentViewModel.cs:105]: Yes

File: NDB.Covid19/NDB.Covid19/Models/SelfDiagnosisSubmissionDTO.cs
Comment (baugztar) [reply NDB.Covid19/NDB.Covid19/Models/SelfDiagnosisSubmissionDTO.cs]: This results in a name conflict with the member from the dto. Updated to reference the type directly.

Comment (pavshr) [reply NDB.Covid19/NDB.Covid19/Models/SelfDiagnosisSubmissionDTO.cs]: Got it

**Review by pavshr — APPROVED** [review]
Good work :)

**Pull request merged**
